src/createServer.js

```diff
@@ -1,3 +1,57 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+/* eslint-disable max-len */
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+const errorMessages = {
+  textError: 'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  caseError: '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+  caseInvalid: 'This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+};
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+    const text = normalizedURL.pathname.slice(1);
+    const toCase = normalizedURL.searchParams.get('toCase');
+    const responseErrors = {
+      errors: [],
+    };
+
+    res.setHeader('Content-Type', 'application/json');
+
+    if (!text) {
+      responseErrors.errors.push({ message: errorMessages.textError });
+    }
+
+    if (!toCase) {
+      responseErrors.errors.push({ message: errorMessages.caseError });
+    }
```

Comment:
Let's Decompose your logic
Move this logic to the separate function
```suggestion
    const ... = ...(...);
```